src/main/java/org/humancellatlas/ingest/protocol/ProtocolRepositoryImpl.java

```diff
@@ -0,0 +1,31 @@
+package org.humancellatlas.ingest.protocol;
+
+import org.humancellatlas.ingest.query.MetadataCriteria;
+import org.humancellatlas.ingest.query.QueryBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Query;
+
+import java.util.List;
+
+public class ProtocolRepositoryImpl implements ProtocolRepositoryCustom {
+    @Autowired
+    private MongoTemplate mongoTemplate;
+
+    @Autowired
+    private QueryBuilder queryBuilder;
+
+    @Override
+    public Page<Protocol> findByCriteria(List<MetadataCriteria> criteriaList, Boolean andCriteria, Pageable pageable) {
+        Query query = queryBuilder.build(criteriaList, andCriteria);
+
+        long count = mongoTemplate.count(query, Protocol.class);
+        query.with(pageable);
+        List<Protocol> result = mongoTemplate.find(query, Protocol.class);
```

Comment:
Or it could be instead add here:

`List<Protocol> result = mongoTemplate.find(query.with(pageable), Protocol.class);`